tests/functions/test_conditional_join.py

```diff
@@ -0,0 +1,807 @@
+    """
+
+    assume(not df.empty)
+    assume(not s.empty)
+    with pytest.raises(TypeError):
+        s.name = "B"
+        df.conditional_join(s, ("B", "B", "<"), suffixes=["_x", "_y"])
+
+
+@given(df=conditional_df(), s=conditional_series())
+def test_check_suffixes_length(df, s):
+    """
+    Raise ValueError if `suffixes` is not a tuple of length 2.
+    """
+    with pytest.raises(ValueError):
+        s.name = "B"
+        df.conditional_join(s, ("B", "B", "<"), suffixes=("_x",))
+        df.conditional_join(s, ("B", "B", "<"), suffixes=("_x", "_y", None))
+
+
+@given(df=conditional_df(), s=conditional_series())
+def test_check_suffixes_None(df, s):
+    """
+    Raise ValueError if `suffixes` is (None, None).
+    """
+    with pytest.raises(ValueError):
+        s.name = "B"
+        df.conditional_join(s, ("B", "B", "<"), suffixes=(None, None))
+
+
+@given(df=conditional_df(), s=conditional_series())
+def test_check_suffixes_subtype(df, s):
+    """
+    Raise TypeError if any entry `suffixes`
+    is not either None or a string type.
+    """
+
+    assume(not df.empty)
+    assume(not s.empty)
+    with pytest.raises(TypeError):
+        s.name = "B"
+        df.conditional_join(s, ("B", "B", "<"), suffixes=(1, None))
+
+
+@given(df=conditional_df())
+def test_check_suffixes_exists_df(df):
+    """
+    Raise ValueError if suffix already exists in `df`.
+    """
+    with pytest.raises(ValueError):
+        right = df.copy()
+        df.columns = ["A", "A_x", "B", "F", "H"]
+        df.conditional_join(right, ("A", "A", "<"), suffixes=("_x", "_y"))
+
+
+@given(df=conditional_df())
+def test_check_suffixes_exists_right(df):
+    """
+    Raise ValueError if suffix already exists in `right`.
+    """
+    with pytest.raises(ValueError):
+        right = df.copy()
+        right.columns = ["A", "B", "B_y", "C", "E"]
+        df.conditional_join(right, ("B", "B", "<"), suffixes=("_x", "_y"))
+
+
+@given(df=conditional_df(), s=conditional_series())
+def test_dtype_Series(df, s):
+    """
+    Raise ValueError if dtype of column in `df`
+    does not match the dtype of column from `right`.
+    """
+    with pytest.raises(ValueError):
+        s.name = "A"
+        df.conditional_join(s, ("C", "A", "<"), suffixes=("_x", "_y"))
+
+
+@given(df=conditional_df(), right=conditional_right())
+def test_single_condition_equality_numeric(df, right):
+    """Test output for a single condition. "=="."""
+    assume(not df.empty)
+    assume(not right.empty)
+    # simulate output as it would be in SQL
+    left_on, right_on = ["B", "Integers"]
+    actual = (
+        df.assign(t=1)
+        .merge(right.assign(t=1), on="t")
+        .query(f"{left_on} == {right_on}")
+        .reset_index(drop=True)
+    )
+    actual = actual.filter([left_on, right_on])
+    expected = df.conditional_join(
+        right, (left_on, right_on, "=="), how="inner", sort_by_appearance=True
+    )
+    expected = expected.filter([left_on, right_on])
```

Comment:
@samukweku I think you might have swapped the names `expected` and `actual`. Just thinking about it from a code reading perspective - in the test suite here, I think line 274 should be `expected` while line 281 should be `actual`, like follows:

```suggestion
    expected = (
        df.assign(t=1)
        .merge(right.assign(t=1), on="t")
        .query(f"{left_on} == {right_on}")
        .reset_index(drop=True)
    )
    expected = expected.filter([left_on, right_on])
    actual = df.conditional_join(
        right, (left_on, right_on, "=="), how="inner", sort_by_appearance=True
    )
    actual = actual.filter([left_on, right_on])
```

Does this match your intuition?